src/embind/embind.js

```diff
@@ -205,14 +205,15 @@ var LibraryEmbind = {
       return body.apply(this, arguments);
     };
 #else
-    /*jshint evil:true*/
-    return new Function(
-        "body",
-        "return function " + name + "() {\n" +
-        "    \"use strict\";" +
-        "    return body.apply(this, arguments);\n" +
-        "};\n"
-    )(body);
+    return {
+      [name]: function(arguments){
+        var context = {};
+        for (var prop in body) {
+          context[prop] = body[prop];
+        };
```

Comment:
Is this the same as `Object.assign(context, body)`?

Is this needed?  Why can't we just use `this` as the context like we do above on line 205?  

Can you remove the `DYNAMIC_EXECUTION == 0` condition above and just always use this method?

Reply:
Oh now I understand.  The following really isn't helpful except to inform you on why I got here.

> Even _string to function or eval()_ lovers should be ok _without the named function_, even, or at least the original... Yes, I'd bet `Object.assign(context, body);` does that for-loop, abstracted.  I would prefer this one, `{ [name]: function(arguments){return body.apply(this,arguments);}.bind(body)}[name]`, also may be an abstracted assign (but declared with it), since the factory 'createNamedFunction' names itself for that purpose and I don't know why else the returned function needs to be named before returning (or why it needs to be named dynamically at all, for that matter). The whole point of this exercise is to declare a wrapper for the call `{[name]:body.apply(body)}[name]`, for [reasons I'm still unsure of](https://stackoverflow.com/questions/9679053/is-it-possible-to-get-the-caller-context-in-javascript/72680938#72680938).  

Do you know why else the createNamedFunction exists?  Your library is versatile but the `DYNAMIC_EXECUTION == 0` gets by without it named, so the whole returned object seems ambivalent to the name... Can it be made into a dummy so [current commands](https://github.com/cloudflare/worker-emscripten-template/blob/master/build.js) don't fail for people?  Leaving that out from the script seems to be @robertaboukhalil idea, leading [us into the evil new Function](https://github.com/robertaboukhalil/cf-workers-emscripten). Maybe some other environments requires that it be named?